src/AutoRest.CSharp/Common/Output/Models/Shared/Parameter.cs

```diff
@@ -1,28 +1,12 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
+#pragma warning disable AD0001 // Doesn't understand record decleration, and the anaylizer throws an exception.
+#pragma warning disable SA1649 // Doesn't understand record decleration, and the anaylizer throws an exception.
```

Comment:
I thought you've updated the analyzer in another PR.

Reply:
Eep - noticed this feedback just after I hit submit.  I did upgrade it in an earlier PR - this was here from before I did that (was trying to work around this issue with this, with no such luck).

I'll remove it in my next set of changes.